tizen-api/include/tizen-api.h

```diff
@@ -0,0 +1,336 @@
+/**
+ * Copyright (c) 2019 Samsung Electronics Co., Ltd All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/**
+ * @file tizen-api.h
+ * @date 07 March 2019
+ * @brief Tizen NNStreamer/Pipeline(main) C-API Header.
+ *        This allows to construct and control NNStreamer pipelines.
+ * @see	https://github.com/nnsuite/nnstreamer
+ * @author MyungJoo Ham <myungjoo.ham@samsung.com>
+ * @bug No known bugs except for NYI items
+ */
+
+#ifndef __TIZEN_NNSTREAMER_API_MAIN_H__
+#define __TIZEN_NNSTREAMER_API_MAIN_H__
+
+#include <tizen_error.h>
+#include <nnstreamer/tensor_typedef.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif /* __cplusplus */
+/**
+ * @addtogroup CAPI_ML_NNSTREAMER_PIPELINE_MODULE
+ * @{
+ */
+
+/**
+ * @brief A handle of an NNStreamer pipeline.
+ * @since_tizen 5.5
+ */
+typedef void *nns_pipeline_h;
+
+/**
+ * @brief A handle of a "sink node" of an NNStreamer pipeline
+ * @since_tizen 5.5
+ */
+typedef void *nns_sink_h;
+
+/**
+ * @brief A handle of a "src node" of an NNStreamer pipeline
+ * @since_tizen 5.5
+ */
+typedef void *nns_src_h;
+
+/**
+ * @brief A handle of a "switch" of an NNStreamer pipeline
+ * @since_tizen 5.5
+ */
+typedef void *nns_switch_h;
+
+/**
+ * @brief A handle of a "valve node" of an NNStreamer pipeline
+ * @since_tizen 5.5
+ */
+typedef void *nns_valve_h;
+
+/**
+ * @brief Enumeration for the error codes of NNStreamer Pipelines.
+ * @since_tizen 5.5
+ * @todo The list is to be filled! (NYI)
+ */
+typedef enum {
+  NNS_ERROR_NONE				= TIZEN_ERROR_NONE, /**< Success! */
+  NNS_ERROR_INVALID_PARAMETER			= TIZEN_ERROR_INVALID_PARAMETER, /**< Invalid parameter */
+  NNS_ERROR_NOT_SUPPORTED			= TIZEN_ERROR_NOT_SUPPORTED, /**< The feature is not supported */
+  NNS_ERROR_PIPELINE_FAIL			= TIZEN_ERROR_STREAMS_PIPE, /**< Cannot create Gstreamer pipeline. */
+} nns_error_e;
+
+/**
+ * @brief Enumeration for buffer deallocation policies.
+ * @since_tizen 5.5
+ * @todo The list is to be filled! (NYI)
+ */
+typedef enum {
+  NNS_BUF_FREE_BY_NNSTREAMER			= 0, /**< Default. Application should not deallocate this buffer. NNStreamer will deallocate when the buffer is no more needed */
```

Comment:
It seems that the length of the line is too long. :)

Reply:
Please note that tizen-api.h is not in the scope of this PR. That part is already merged.